src/System.Private.CoreLib/shared/System/Text/UTF32Encoding.cs

```diff
@@ -385,7 +387,7 @@ internal override unsafe int GetByteCount(char* chars, int count, EncoderNLS enc
 
                 // We mustn't have left over fallback data when counting
                 if (fallbackBuffer.Remaining > 0)
-                    throw new ArgumentException(SR.Format(SR.Argument_EncoderFallbackNotEmpty, this.EncodingName, encoder.Fallback.GetType()));
+                    throw new ArgumentException(SR.Format(SR.Argument_EncoderFallbackNotEmpty, this.EncodingName, encoder.Fallback!.GetType())); // TODO-NULLABLE: NullReferenceException
```

Comment:
This one is already on an exceptional path; let's just fix it, e.g. `encoder.Fallback?.GetType() ?? string.Empty`.  Same for any others here that are part of building an exception.